### PR TITLE
Various additions

### PR DESCRIPTION
**Pull request by aymgal**
- Effective slope and Einstein radius returns NaN : slight modification of the implementation in `LensModelExtensions` since I think it is easier to handle NaNs than a (wrong) real value when trying to compute those effective quantities. No change in the interface though, and an additional setting to remove warning messages.

- Numerical method for computing velocity dispersion now can return MGE model lists and related parameter kwargs for checking MGE accuracy. Also no change in the interface.

### CODE REVIEW

File: lenstronomy/Analysis/lens_properties.py [hunk 1]
```diff
@@ -166,6 +166,8 @@ def velocity_dispersion_numerical(self, kwargs_lens, kwargs_lens_light, kwargs_a
                         anisotropy_model=anisotropy_model, fwhm=psf_fwhm, psf_type=psf_type, moffat_beta=moffat_beta,
                         kwargs_cosmo=kwargs_cosmo, **kwargs_numerics)
         sigma2 = galkin.vel_disp(kwargs_profile, kwargs_light, kwargs_anisotropy, kwargs_aperture)
+        if return_MGE_results:
```
Comment (sibirrer) [inline lenstronomy/Analysis/lens_properties.py]: I don't like options that change the return product of a function. This indicates that you want to have better access to underlying statements with a different function. I just pushed a small change that separates the two computations or at least can give you conveniently access to the statements that you (optionally) are requesting.

File: test/test_Analysis/test_lens_properties.py [hunk 2]
```diff
@@ -62,6 +62,18 @@ def test_velocity_dispersion_new(self):
         npt.assert_almost_equal(v_sigma_mge_lens / v_sigma, 1, decimal=1)
         npt.assert_almost_equal(v_sigma / v_sigma_hernquist, 1, decimal=1)
 
+        # test with MGE results returned
+        output = lensProp.velocity_dispersion_numerical(kwargs_lens, kwargs_lens_light, kwargs_anisotropy, kwargs_aperture,
```
Comment (sibirrer) [inline test/test_Analysis/test_lens_properties.py]: see my comments above. I'll add a specific test to the new function that I provide that should make this redundant.

File: lenstronomy/LensModel/lens_model_extensions.py [hunk 3]
```diff
@@ -316,7 +317,11 @@ def profile_slope(self, kwargs_lens_list, lens_model_internal_bool=None, num_poi
         :param num_points: number of estimates around the Einstein radius
         :return:
         """
-        theta_E = self.effective_einstein_radius(kwargs_lens_list)
+        theta_E = self.effective_einstein_radius(kwargs_lens_list, verbose=verbose)
+        if np.isnan(theta_E):
```
Comment (sibirrer) [inline lenstronomy/LensModel/lens_model_extensions.py]: are these lines tested?

File: lenstronomy/LensModel/lens_model_extensions.py [hunk 4]
```diff
@@ -240,8 +240,9 @@ def effective_einstein_radius(self, kwargs_lens_list, k=None,
                         return r, r_array[1] - r_array[0]
                     else:
                         return r
-        print(kwargs_lens_list, "Warning, no Einstein radius computed!")
-        return r_array[-1]
+        if verbose:
```
Comment (sibirrer) [inline lenstronomy/LensModel/lens_model_extensions.py]: are these lines tested?

**Comment by aymgal** [comment]
I added the missing tests and solve conflicts in the last commit

File: lenstronomy/Analysis/lens_properties.py [hunk 5]
```diff
@@ -78,7 +78,7 @@ def velocity_dispersion(self, kwargs_lens, r_eff, R_slit, dR_slit, psf_fwhm, ani
     def velocity_dispersion_numerical(self, kwargs_lens, kwargs_lens_light, kwargs_anisotropy, kwargs_aperture, psf_fwhm,
                                       aperture_type, anisotropy_model, r_eff, psf_type='GAUSSIAN', moffat_beta=2.6, kwargs_numerics={}, MGE_light=False,
                                       MGE_mass=False, lens_model_kinematics_bool=None, light_model_kinematics_bool=None,
-                                      Hernquist_approx=False):
+                                      Hernquist_approx=False, return_MGE_results=False):
```
Comment (sibirrer) [inline lenstronomy/Analysis/lens_properties.py]: is this still a function argument that is used?

Comment (aymgal) [reply lenstronomy/Analysis/lens_properties.py]: Forgot to delete that, it's done now

**Comment by sibirrer** [comment]
Thank you!

**Pull request merged**